src/org/skypro/skyshop/search/SearchEngine.java

```diff
@@ -1,28 +1,27 @@
 package org.skypro.skyshop.search;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.Comparator;
 
 public class SearchEngine {
-    private final List<Searchable> items;
+    private final Set<Searchable> items;
 
     public SearchEngine() {
-        items = new ArrayList<>();
+        items = new HashSet<>();
     }
 
     public void add(Searchable item) {
         items.add(item);
     }
 
-    // Метод поиска теперь возвращает отсортированную Map с ключами - именами
-    public Map<String, Searchable> search(String term) {
-        Map<String, Searchable> results = new TreeMap<>();
-
+    public SortedSet<Searchable> search(String term) {
```

Comment:
не забываем про полиморфизм
```suggestion
    public Set<Searchable> search(String term) {
```